GlCanvas.js

```diff
@@ -0,0 +1,261 @@
+(function(global, $) {
+function GlPlotter() {
+    this.pluginName = 'glplotter';
+    this.pluginVersion = '0.2';
+    /**
+     * Initialize a webgl layer
+     */
+    this.init = function(plot) {
+        var defaultPlotOffset = {
+            left: 0,
+            right: 0,
+            top: 0,
+            bottom: 0
+        };
+
+        var textures = [],
+            materials = [],
+            geometries = [];
+        var container, webglsurface, canvas, width, height,
+            renderer;
+        plot.hooks.processOptions.push(processOptions);
+
+        function processOptions(plot, options) {
+            var mainScene, camera, cameraFocus;
+            container = plot.getPlaceholder();
+            webglsurface = plot.getWebGlSurface();
+            canvas = plot.getWebGlCanvas();
+            width = webglsurface.width;
+            height = webglsurface.height;
+            renderer = webglsurface.renderer;
+            if(!renderer.userData) {
```

Comment:
I think it's possible the renderer to be null/undefined which will throw an exception.